src/compiler/checker.ts

```diff
@@ -16430,6 +16440,24 @@ namespace ts {
             return getSignatureInstantiation(signature, getInferredTypes(context), isInJavaScriptFile(contextualSignature.declaration));
         }
 
+        function inferJsxTypeArguments(signature: Signature, sourceAttributesType: Type, context: InferenceContext): Type[] {
+            // Clear out all the inference results from the last time inferTypeArguments was called on this context
+            for (const inference of context.inferences) {

```

Comment:
when is a context reused? I only see inferJsxTypeArguments being called right after creating a fresh context.

Reply:
Good point. It may be possible to reuse them in some way in the future, but I'm not doing so now.